drivers/spi/spi_sy1xx.c

```diff
@@ -0,0 +1,499 @@
+#define SPI_CMD_EOT       (9 << SPI_CMD_OFFSET)
+#define SPI_CMD_RPT_END   (10 << SPI_CMD_OFFSET)
+#define SPI_CMD_RX_CHECK  (11 << SPI_CMD_OFFSET)
+#define SPI_CMD_FULL_DPLX (12 << SPI_CMD_OFFSET)
+
+/* CMD CFG */
+#define SPI_CMD_CFG_0()           (SPI_CMD_CFG)
+#define SPI_CMD_CFG_1()           (0)
+#define SPI_CMD_CFG_2(cpol, cpha) (((cpol & 0x1) << 1) | ((cpha & 0x1) << 0))
+#define SPI_CMD_CFG_3(div)        (div & 0xff)
+
+/* CMD SOT */
+#define SPI_CMD_SOT_0()   (SPI_CMD_SOT)
+#define SPI_CMD_SOT_1()   (0)
+#define SPI_CMD_SOT_2()   (0)
+#define SPI_CMD_SOT_3(cs) (cs & 0x1)
+
+/* CMD SEND_ADDR */
+#define SPI_CMD_SEND_ADDR0(qspi)     (SPI_CMD_SEND_ADDR | ((qspi & 0x1) << 3))
+#define SPI_CMD_SEND_ADDR1(num_bits) ((num_bits - 1) & 0x1f)
+#define SPI_CMD_SEND_ADDR2()         (0)
+#define SPI_CMD_SEND_ADDR3()         (0)
+
+/* CMD SEND_DATA */
+#define SPI_CMD_SEND_DATA0(qspi)     (SPI_CMD_TX_DATA | ((qspi & 0x1) << 3))
+#define SPI_CMD_SEND_DATA1()         (0)
+#define SPI_CMD_SEND_DATA2(num_bits) (UINT16_BYTE0((num_bits - 1)) & 0xffff)
+#define SPI_CMD_SEND_DATA3(num_bits) (UINT16_BYTE1((num_bits - 1)) & 0xffff)
+
+/* CMD READ_DATA */
+#define SPI_CMD_READ_DATA0(qspi, align)                                                            \
+	(SPI_CMD_RX_DATA | ((qspi & 0x1) << 3) | ((align & 0x3) << 1))
+#define SPI_CMD_READ_DATA1()         (0)
+#define SPI_CMD_READ_DATA2(num_bits) (UINT16_BYTE0((num_bits - 1)) & 0xffff)
+#define SPI_CMD_READ_DATA3(num_bits) (UINT16_BYTE1((num_bits - 1)) & 0xffff)
+
+/* CMD FULL_DPLX */
+#define SPI_CMD_FULL_DPLX_DATA0(qspi, align)                                                       \
+	(SPI_CMD_FULL_DPLX | ((qspi & 0x1) << 3) | ((align & 0x3) << 1))
+#define SPI_CMD_FULL_DPLX_DATA1()         (0)
+#define SPI_CMD_FULL_DPLX_DATA2(num_bits) (UINT16_BYTE0((num_bits - 1)) & 0xffff)
+#define SPI_CMD_FULL_DPLX_DATA3(num_bits) (UINT16_BYTE1((num_bits - 1)) & 0xffff)
+
+/* CMD EOT */
+#define SPI_CMD_EOT0()    (SPI_CMD_EOT)
+#define SPI_CMD_EOT1()    (0)
+#define SPI_CMD_EOT2()    (0)
+#define SPI_CMD_EOT3(evt) (evt & 0x1)
+
+/* CMD Wait */
+#define SPI_CMD_WAIT0() (SPI_CMD_DUMMY)
+#define SPI_CMD_WAIT1() (0xff)
+#define SPI_CMD_WAIT2() (0xff)
+#define SPI_CMD_WAIT3() (0xff)
+
+/* Hardware chip select dt slave fields; set reg, if hw chip select shall be used */
+#define SY1XX_CS_HW_SELECT_0 0x80
+#define SY1XX_CS_HW_SELECT_1 0x81
+
+#define UINT16_BYTE0(uint16) ((uint16 >> 8) & 0xff)
+#define UINT16_BYTE1(uint16) ((uint16 >> 0) & 0xff)
+
+#define SY1XX_SPI_MIN_FREQUENCY   (250000)
+#define SY1XX_SPI_MAX_FREQUENCY   (62500000)
+#define SY1XX_SPI_WORD_SIZE_8_BIT (8)
+#define SY1XX_SPI_WORD_ALIGN      (0)
+#define SY1XX_SPI_MAX_BIT_COUNT   (10240)
+#define SY1XX_SPI_MAX_BUFFER_SIZE (SY1XX_SPI_MAX_BIT_COUNT / 8)
+
+struct sy1xx_spi_config {
+	/* dma base address */
+	uint32_t base;
+	/* number of instance, spi0, spi1, ... */
+	uint32_t inst;
+	/* quad spi enabled */
+	uint8_t quad_spi;
```

Comment:
That's useless in implementing spi.h api. Dual, quad and over are not supported there